homeassistant/components/telegram_bot/__init__.py

```diff
@@ -286,7 +286,8 @@ def load_data(
             _LOGGER.warning("Can't load data in %s after %s retries", url, retry_num)
         elif filepath is not None:
             if hass.config.is_allowed_path(filepath):
-                return open(filepath, "rb")  # pylint: disable=consider-using-with
+                with open(filepath, "rb") as fp:
+                    return fp
```

Comment:
Will this close the file after the return? Does that change how this works?

Reply:
Good catch. I missed that we only return the file handler itself which still needs to be open. Thus the change would result in an error. I mixed it up with `return fp.read()` which would have worked.

Reply:
Opened: https://github.com/PyCQA/pylint/issues/4748